traces/traces.go

```diff
@@ -0,0 +1,101 @@
+package traces
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	instrumentationPkg = "github.com/osquery/osquery-go"
+	osqueryGoVersion   = "0.0.0"
+)
+
+// OsqueryGoTracer provides a tracer with a standardized name and version.
+func OsqueryGoTracer() trace.Tracer {
```

Comment:
What uses this?

Reply:
Nothing at the moment -- I can remove it if we don't think it'd be useful.

Reply:
What's the intent? I don't really understand how it's useful

Reply:
If an area in osquery-go needed to start a span with `SpanStartOption`s not supported by `StartSpan` below, we could use `traces.OsqueryGoTracer().Start(ctx, "some-span-name", customSpanOptions)` and that would guarantee a tracer that uses the correct instrumentation package name `github.com/osquery/osquery-go`.

Also I realized I should probably be using this for consistency on line 51 below too, instead of repeating `otel.Tracer(instrumentationPkg, trace.WithInstrumentationVersion(osqueryGoVersion))`.